_examples/json/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/alecthomas/kong"
+	"github.com/alecthomas/participle"
+)
+
+type pathExpr struct {
+	Parts []part `@@ { "." @@ }`
+}
+
+type part struct {
+	Obj string `@Ident`
+	Acc []acc  `("[" @@ "]")*`
+}
+
+type acc struct {
+	Name  *string `@(String|Char|RawString)`
+	Index *int    `| @Int`
+}
+
+var (
+	parser = participle.MustBuild(&pathExpr{})
+	cli    struct {
+		File string `arg:"" type:"existingfile" help:"File to parse."`
+	}
+)
+
+func main() {
+	ctx := kong.Parse(&cli)
+	r, err := os.Open(cli.File)
+	ctx.FatalIfErrorf(err)
+	defer r.Close()
+
+	var input map[string]interface{}
+	err = json.NewDecoder(r).Decode(&input)
+	ctx.FatalIfErrorf(err)
+
+	enc := json.NewEncoder(os.Stdout)
+	enc.SetIndent("", "  ")
+
+	for {
+		// Example: check_run.check_suite.pull_requests[0].head.sha
+		fmt.Fprint(os.Stderr, "Enter a json path: ")
```

Comment:
I'd just make this an argument ge. `json <query> <files...>`

Reply:
I made it so it's not interactive any more. Instead you pass in a query and 1+ files